src/wcmConfig.c

```diff
@@ -1079,4 +1083,162 @@ void wcmDevClose(WacomDevicePtr priv)
 	}
 }
 
+#ifdef ENABLE_TESTS
+
+/**
+ * After a call to wcmInitialToolSize, the min/max and resolution must be
+ * set up correctly.
+ *
+ * wcmInitialToolSize takes the data from the common rec, so test that the
+ * priv has all the values of the common.
+ */
+TEST_CASE(test_initial_size)
+{
+	WacomDeviceRec priv = {0};
+	WacomCommonRec common = {0};
+
+	/* pin to some numbers */
+	int xres = 1920, yres = 1600;
+	int minx, maxx = 2 * xres, miny, maxy = 2 * yres;
+
+	priv.common = &common;
+
+	/* FIXME: we currently assume min of 0 in the driver. we cannot cope
+	 * with non-zero devices */
+	minx = miny = 0;
+
+	common.wcmMaxX = maxx;
+	common.wcmMaxY = maxy;
+	common.wcmResolX = xres;
+	common.wcmResolY = yres;
+
+	wcmInitialToolSize(&priv);
+
+	assert(priv.topX == minx);
+	assert(priv.topY == minx);
```

Comment:
This line has a typo on it that I was going to fix in a separate pull request. You can ignore it for now, but I thought I'd mention it.

Reply:
oooh, an almost 11y old bug! congratulations! :)

Fixed in a follow-up commit, thanks!